docs/adrs/00003-external-references.md

```diff
@@ -0,0 +1,1246 @@
+        "pkg:rpm/redhat/openssl-debugsource@3.0.7-18.el9_2?arch=x86_64",
+        "pkg:rpm/redhat/openssl-devel@3.0.7-18.el9_2?arch=aarch64",
+        "pkg:rpm/redhat/openssl-devel@3.0.7-18.el9_2?arch=i686",
+        "pkg:rpm/redhat/openssl-devel@3.0.7-18.el9_2?arch=ppc64le",
+        "pkg:rpm/redhat/openssl-devel@3.0.7-18.el9_2?arch=s390x",
+        "pkg:rpm/redhat/openssl-devel@3.0.7-18.el9_2?arch=x86_64",
+        "pkg:rpm/redhat/openssl-libs@3.0.7-18.el9_2?arch=aarch64",
+        "pkg:rpm/redhat/openssl-libs@3.0.7-18.el9_2?arch=i686",
+        "pkg:rpm/redhat/openssl-libs@3.0.7-18.el9_2?arch=ppc64le",
+        "pkg:rpm/redhat/openssl-libs@3.0.7-18.el9_2?arch=s390x",
+        "pkg:rpm/redhat/openssl-libs@3.0.7-18.el9_2?arch=x86_64",
+        "pkg:rpm/redhat/openssl-libs-debuginfo@3.0.7-18.el9_2?arch=aarch64",
+        "pkg:rpm/redhat/openssl-libs-debuginfo@3.0.7-18.el9_2?arch=i686",
+        "pkg:rpm/redhat/openssl-libs-debuginfo@3.0.7-18.el9_2?arch=ppc64le",
+        "pkg:rpm/redhat/openssl-libs-debuginfo@3.0.7-18.el9_2?arch=s390x",
+        "pkg:rpm/redhat/openssl-libs-debuginfo@3.0.7-18.el9_2?arch=x86_64",
+        "pkg:rpm/redhat/openssl-perl@3.0.7-18.el9_2?arch=aarch64",
+        "pkg:rpm/redhat/openssl-perl@3.0.7-18.el9_2?arch=i686",
+        "pkg:rpm/redhat/openssl-perl@3.0.7-18.el9_2?arch=ppc64le",
+        "pkg:rpm/redhat/openssl-perl@3.0.7-18.el9_2?arch=s390x",
+        "pkg:rpm/redhat/openssl-perl@3.0.7-18.el9_2?arch=x86_64"
+      ]
+    }
+  ]
+}
+```
+
+A single PACKAGE_OF relationship is established between product sbom library component and the equiv component in component sbom.
+
+It is the case that **evidence.identity** in product sbom == **evidence.identity** in component sbom ... the hashes should also be equiv.
+
+The result of querying for deps would then resolve PACKAGE_OF relationship and then expand downwards from component sbom.
+
+We should use the same internal mapping as we do other external references, for example: 
+
+   `urn:uuid:6873e671-5b39-f541-0fa3-1a69b79a5892/1#openssl-3.0.7-18.el9_2comnponentsbom`
+
+Would be used to create an external reference.
+
+If `bom-ref` does not exist we can assist heuristic by generating an implicit `bom-ref` (ex. assume bom-ref = purl).
+
+### Handling errors and asynchronous ingestion
+
+When we ingest an SBOM that refers to another SBOM - we establish the potential for a relationship. 
+
+It is only at 'query time' when we try to resolve the relationship - if an external SBOM does not 
+exist (eg. never ingested, error, etc) then we should raise an appropriate error.
+
+## Decision
+
+### Implement **sbom_external_node** table
+
+- documentNampace (url)
+- document version
+- Checksum/digest/hash
+- conflated reference
+
+Then a node, in `sbom_node` table, can use the conflated reference: 
+
+    _pkg:rpm/redhat/openssl@3.0.7-18.el9_2?arch=src_ **PACKAGES** _https://example.org/other-sbom.json#openssl-3.0.7-18.el9_2_ 
```

Comment:
This line confuses me. What is `**PACKAGES**`?

Can we add a section to this doc that shows the exact data for a row in the new table along with the relevant sections of the two SBOM's?

Reply:
this is a result of docs/adrs/00002-analysis-graph.md eg PACKAGE_OF reverses relationship to PACKAGES ... we want to create a tree with only one direction relationships.